### PR TITLE
Full revision of rid catalog topic for 2.0

### LINKED ISSUE #2843: Add information about portable RIDs
With .NET Core 2.0 we are moving towards using what we call portable RID's which are essentially RID's that aren't distro or version specific but instead contain assets that can work on lots of platforms. I think we need to refresh this document https://docs.microsoft.com/en-us/dotnet/core/rid-catalog to add information on the portable RIDs (i.e. linux-x64, win-x86/x64, osx-x64, etc). 

cc @ericstj

### PR DESCRIPTION
**Pull request by mairaw**
Fixes #2843 

[Internal Review URL](https://review.docs.microsoft.com/en-us/dotnet/core/rid-catalog?branch=pr-en-us-3079)

### CODE REVIEW

File: docs/core/rid-catalog.md [hunk 1]
```diff
@@ -1,136 +1,176 @@
...
-system with varying platform API surface area.
-- `[arch]` is the processor architecture, for example, `x86`, `x64`, `arm`, `arm64`, etc.
-
-The RID graph is defined in a package called `Microsoft.NETCore.Platforms` in a file called `runtime.json`, which you can 
-see on the [CoreFX repo](https://github.com/dotnet/corefx/blob/master/pkg/Microsoft.NETCore.Platforms/runtime.json). If 
-you use this file, you will notice that some of the RIDs have an `"#import"` statement in them. These statements are 
-compatibility statements. That means that a RID that has an imported RID in it can be a target for restoring packages 
-for that RID. Slightly confusing, but let's look at an example. Let's take a look at macOS:
+RID is short for *Runtime IDentifier*. RID values are used to identify target platforms where the application runs.
+They're used by .NET packages to represent runtime-specific assets in NuGet packages. The following values are examples of RIDs: `linux-x64`, `ubuntu.14.04-x64`, `win7-x64`, or `osx.10.11-x64`.
```
Comment (Petermarcu) [inline docs/core/rid-catalog.md]: Its not "runtime specific assets", it's "platform specific assets".

File: docs/core/rid-catalog.md [hunk 2]
```diff
@@ -1,136 +1,176 @@
...
+- `[additional qualifiers]` further differentiate different platforms. For example: `aot` or `corert`.
+
+## RID graph
+
+The RID graph or runtime fallback graph is a list of RIDs that are compatible with each other. The RIDs are defined in the [Microsoft.NETCore.Platforms](https://www.nuget.org/packages/Microsoft.NETCore.Platforms/) package. You can see the list of supported RIDs and the RID graph in the [*runtime.json*](https://github.com/dotnet/corefx/blob/master/pkg/Microsoft.NETCore.Platforms/runtime.json) file, which is located at the CoreFX repo. In this file, you can see that all RIDs, except for the base one, contain an `"#import"` statement. These statements indicate compatible RIDs.
+
+When NuGet restores packages, it tries to find an exact match for the specified runtime.
+If an exact match is not found, NuGet walks back the graph until it finds the closest compatible system according to the RID graph.
+
+The following example is the actual entry for the `osx.10.11-x64` RID:
```
Comment (Petermarcu) [inline docs/core/rid-catalog.md]: May be better to use a platform that is still supported? 10.12?

File: docs/core/rid-catalog.md [hunk 3]
```diff
@@ -1,136 +1,176 @@
...
+- The RIDs need to be specific, so don't assume anything from the actual RID value.
 
 ## Using RIDs
-In order to use RIDs, you have to know which RIDs there are. New RIDs are added regularly to the platform. 
-For the latest version, please check the [runtime.json](https://github.com/dotnet/corefx/blob/master/pkg/Microsoft.NETCore.Platforms/runtime.json) file on CoreFX repo.
 
-> [!NOTE]
-> We are working towards getting this information into a more interactive form. 
-> When that happens, this page will be updated to point to that tool and/or its usage documentation. 
+To be able to use RIDs, you have to know which RIDs exist. New values are added regularly to the platform.
```
Comment (Petermarcu) [inline docs/core/rid-catalog.md]: There is a way for a package to extend the graph if needed. Should we doc that?

Comment (ericstj) [reply docs/core/rid-catalog.md]: @robrrelyea / @emgarten?  I recall that previous NuGet owners didn't want to doc this.  Now with the prevalence of distros we may need to.

Comment (emgarten) [reply docs/core/rid-catalog.md]: Dependencies through runtime.json is considered something that only system packages should use. I'm not sure on the RID graph itself which is also part of runtime.json.

Comment (mairaw) [reply docs/core/rid-catalog.md]: I'm a bit confused on what exactly to do here. cc @rrelyea

**Comment by rrelyea** [comment]
Let's make this part of Netcore 2.1 planning process if this change is a significantly important change to ask of Nuget. Would have to balance with other asks.

File: docs/core/rid-catalog.md [hunk 4]
```diff
@@ -1,136 +1,176 @@
...
+  - `linux-x64`
+- CentOS
+  - `centos-x64`
+  - `centos.7-x64`
+- Debian
+  - `debian-x64`
+  - `debian.8-x64`
+- Fedora
+  - `fedora-x64`
+  - `fedora.23-x64`
```
Comment (eerhardt) [inline docs/core/rid-catalog.md]: I think we can remove fedora.23 and opensuse.13.2 since they've been "end of life'd".

**Review by eerhardt — APPROVED** [review]
Looks good.  Thanks for the good documentation @mairaw.

**Review by weshaggard — APPROVED** [review]

**Pull request merged**